apps/passport-server/test/devconnect.spec.ts

```diff
@@ -373,6 +384,234 @@ describe("devconnect functionality", function () {
     ]);
   });
 
+  /**
+   * This test shows the case where a ticket has been checked in with
+   * Pretix, but not in PCDPass. The ticket will be marked as consumed
+   * on the basis of data received from Pretix.
+   */
+  step("should be able to sync a checked-in ticket", async function () {
+    const devconnectPretixAPIConfigFromDB = await getDevconnectPretixConfig(db);
+    if (!devconnectPretixAPIConfigFromDB) {
+      throw new Error("Could not load API configuration");
+    }
+
+    const organizer = devconnectPretixAPIConfigFromDB?.organizers[0];
+    const orgUrl = organizer.orgURL;
+
+    // Pick an event where we will consume all of the tickets
+    const eventID = organizer.events[0].eventID;
+    const eventConfigID = organizer.events[0].id;
+    const org = mocker.get().organizersByOrgUrl.get(orgUrl) as IOrganizer;
+
+    // Simulate Pretix returning tickets as being checked in
+    server.use(
+      rest.get(orgUrl + `/events/:event/orders`, (req, res, ctx) => {
+        const skip = (req.params.event as string) !== eventID;
+        const orders = skip
```

Comment:
strongly typing this would have prevented this problem I think https://github.com/proofcarryingdata/zupass/pull/586/

Reply:
Added typing on https://github.com/proofcarryingdata/zupass/pull/586

Reply:
This would have surfaced a conflict between the fact that I was using a `string` here, despite the typing being `Date`. However, I would have just used a `Date` instead, so it wouldn't have produced any difference in the final output (it all gets JSON.stringified a few lines down, and `JSON.stringify(dateObject) === JSON.stringify(dateObject.toISOString())`.

The thing causing a real problem is the bit where I assumed the inverse, that `JSON.parse(dateString)` produces a `Date`, which is obviously stupid. I was able to make that mistake because we do the equivalent of `JSON.parse(pretixResponse) as SomeType` without actually checking if the data conforms to the type. This is fine so long as our types are only JSON-compatible primitives and the types really do line up to the Pretix API, but in this case I introduced a `Date` object and then didn't write the appropriate custom code for parsing JSON strings as dates.